[16.0][MIG] partner_contact_in_several_companies: Migration to 16.0

stephansainleger opened the pull request:
new PR replacing PR #1703 (wrong git manipulation...)

Commit:
Deprecate base_contact.

- Remove nationality and birthday features.
- Move the remaining features to partner_contact_in_several_companies.
- Make base_contact a dummy module.
- Warn deprecation everywhere possible.

sysadminmatmoz committed:
Translations and templates sync

yvaucher committed:
Add missing default oca icons

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

Commit:
OCA module upgrade to version 9.0.

- Remove Domain Error on contact view.
- Additionally, improve behaviour of personal contact info page.
- Change to README.
- No longer dependent on partner_contact_personal_information_page
- Added tests for ir_actions
- Include extra demo data.
- Increase test coverage.
- Remove birthdate from demo data.

yvaucher committed:
Making the 'personal contact' informations depentens of the module 'personal_contact_information'

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

Commit:
[MIG] 10.0 partner_contact_in_several_companies (#347)

* [MIG] inital work to 10.0

* [FIX] restore computed field

- select=1 is not usable anymore ...

* [FIX] convert _fields_sync to new api

* [FIX] restore form . import statement

* [FIX] readme with last template.

* [FIX] import statement.

* [FIX] check all tests

* [FIX] Flake8 errors fixes

* [FIX] flake 8

* [FIX] wrong obj call self -> self.contact_id

* [FIX] typo

* [FIX] flake 8

* [FIX] add index on contact_type

* [fix] use in statement on test

* [FIX] update base action with new context

* [FIX] test fixes

* [FIX] test work

* [FIX] flake 8 over indent

* [FIX] fixes on lastest comments

- better readability
- remove <data></data> tags

* [FIX] model that same pattern for _fields_sync

- add test ensure_once()

* [FIX] remove last data xml tags

* [FIX] fix action test

* [FIX] typo in import statement

* [FIX] remove logger from tests

* [FIX] new flake8 compute method name for OCA

* [FIX] minor typo

* [FIX] partner_contact_in_several_companies: remove contributor key in manifest

* [FIX] partner_contact_in_several_companies: residual onchange on partner view

* Revert "[FIX] partner_contact_in_several_companies: remove contributor key in manifest"

This reverts commit 56e511a9a7e1dd20cf99578a340a00301a98ba85.

* [FIX] partner_contact_personal_information_page: clean manifest and authors

* [FIX] partner_contact_in_several_companies: required = True not needed

Commit:
[MIG] V11 migration

- fax field removed from res.partner
- 2to3 on all py files
- add contributor and update README
- use hasclass in place of @class in xpath expr
- improve tests

[FIX] typo and clean README

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] partner_contact_in_several_companies: Incorrect syntax for XML context

drdran committed:
[FIX] partner_contact_in_several_companies: Fix problem in view (#541)

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[FIX] disable position editing on contact

- This is just to see all positions
- If we want to add position go to one company and add contact linked to this partner

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-11.0/partner-contact-11.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-11-0/partner-contact-11-0-partner_contact_in_several_companies/es/

kongrattapong committed:
[MIG] partner_contact_in_several_companies: Migration to 12.0

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

OCA-git-bot committed:
partner_contact_in_several_companies 12.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-12.0/partner-contact-12.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_contact_in_several_companies/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_contact_in_several_companies/de/

foliveira-odoogap committed:
[MIG] partner_contact_in_several_companies: Migration to 13.0

foliveira-odoogap committed:
[IMP] fix minor changes in manyfest and view

foliveira-odoogap committed:
[IMP] check travis failure

foliveira-odoogap committed:
[IMP] check travis failure_2

foliveira-odoogap committed:
[IMP] check travis failure_3

foliveira-odoogap committed:
[IMP] pre-commit version downgrade

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

OCA-git-bot committed:
partner_contact_in_several_companies 13.0.1.1.0

jacob-willdooit committed:
[14.0][MIG] partner_contact_in_several_companies

jacob-willdooit committed:
PR - Requested changes

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

OCA-git-bot committed:
partner_contact_in_several_companies 14.0.1.1.0

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 77.7% (28 of 36 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/fr_FR/

sebalix committed:
[FIX] Fix access right issues when using actions

OCA-git-bot committed:
partner_contact_in_several_companies 14.0.1.1.1

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 20.6% (6 of 29 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/it/

klodr committed:
Translated using Weblate (French (France))

Currently translated at 77.7% (28 of 36 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/fr_FR/

klodr committed:
Translated using Weblate (French (Switzerland))

Currently translated at 3.4% (1 of 29 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/fr_CH/

klodr committed:
Translated using Weblate (French)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/fr/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 20.6% (6 of 29 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (36 of 36 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_contact_in_several_companies/es/

Commit:
[IMP] partner_contact_in_several_companies: black, isort, prettier

Commit:
[MIG] partner_contact_in_several_companies: Migration to 16.0

Commit:
[IMP] partner_in_several_companies: complete README.rst

stephansainleger commented:
Hi @legalsylvain !
After several misadventures with Git (including the unintentional closure of PR [#1703](https://github.com/OCA/partner-contact/pull/1703) ...), I was finally able to properly follow the migration procedure for the module partner_contact_in_several_companies.
Could you please let me know if everything is OK so we can finally complete this migration?
Thank you!

PicchiSeba commented:
I think we should take advantage of this occasion to comply with OCA module documentation.

The file README.md added in 9d8a0b2f093ffae0e56436d1598be49dea45d9a4 should be generated automatically.
To do so we should move it to a directory called `readme` and change the extension from `.md` to `.rst`.
In the same directory we should create two files called `CONTRIBUTORS.rst` and `USAGE.rst`, in the latter we must move the "Usage" section from README.rst

Then we can regenerate the "README.rst" file in the root folder with `oca-gen-addon-readme`

HekkiMelody commented on partner_contact_in_several_companies/models/res_partner.py:
```diff
@@ -0,0 +1,158 @@
+            args, offset=offset, limit=limit, order=order, count=count
+        )
+
+    @api.model
+    def create(self, vals):
+        """When creating, use a modified self to alter the context (see
+        comment in _basecontact_check_context).  Also, we need to ensure
+        that the name on an attached contact is the same as the name on the
+        contact it is attached to."""
+        modified_self = self._basecontact_check_context("create")
+        if not vals.get("name") and vals.get("contact_id"):
+            vals["name"] = modified_self.browse(vals["contact_id"]).name
+        return super(ResPartner, modified_self).create(vals)
```
**issue**: When I install the module, I have this warning:

```
WARNING devel odoo.api.create: The model odoo.addons.partner_contact_in_several_companies.models.res_partner is not overriding the create method in batch 
```

It's probably enough to just change the decorator to `@api.model_create_multi`

stephansainleger replied:
Thanks @aleuffre for the notice, I have corrected this issue: https://github.com/OCA/partner-contact/pull/1852/files#diff-bc484720a893a4002fc161049b55f9f26bdc7a310e9dfdc656a5d32c7ce9a3c1R74

HekkiMelody reviewed (changes requested):
Hello, thank you for the PR!

Just a small warning

rousseldenis commented:
/ocabot migration partner_contact_in_several_companies

tobru commented:
Thank you so much for taking the time to migrate this add-on to Odoo 16. I just gave this PR a try in our Odoo 16, and it worked well so far. What I have seen is that in a Standalone contact in the "Other Positions" tab, the formatting seems to have some issues:

![Screenshot_20241104_170303](https://github.com/user-attachments/assets/4a66da2f-c415-4983-9f26-276853ef14b8)

* There is an artificial "Avatar" showing, I'd expect either nothing or a dummy avatar
* The line above the blurred name is empty and the "x" is placed at a strange place

Additionally, when clicking on this contact in the "Other Positions" tab, the pop-up is not really helpful, there is no mentioning to which other record this contact is related to. But maybe this was already the case in older versions and not related to the migration.

![Screenshot_20241104_170644](https://github.com/user-attachments/assets/2449310d-7b54-4659-be1f-a6ae771ebee7)

stephansainleger commented:
@tobru Thanks for your remarks. I have corrected:
- the image issue
- the cross (I have removed it as it comes from older versions and is not consistent anymore with similar displays)
For the pop-up, I prefer not to modify it now, as it is a deeper change not linked with the migration :-)

HekkiMelody reviewed (approved):
Code review, LGTM

stephansainleger commented:
Hi @pedrobaeza ! 
I read [here](https://oca.github.io/repo-maintainer-conf/teams.html#crm-sales-marketing-maintainers-psc-representative) that you might be the guy to tag for this PR final integration. Is that correct? 
Thanks ! :-)

pedrobaeza reviewed (dismissed):
Please don't have README.md and README.rst. It should be README.rst for proper parsing inside Odoo, and you can make use of the fragments inside `readme` folder for letting the pre-commit hook to generate a complete README.rst file.

stephansainleger commented:
> Please don't have README.md and README.rst. It should be README.rst for proper parsing inside Odoo, and you can make use of the fragments inside `readme` folder for letting the pre-commit hook to generate a complete README.rst file.

Thanks @pedrobaeza ! I have pushed a new `READED.rst` and removed the `.md`. I had issues with pre-commit locally, so I still only have the README.rst file, without the `readme` repo. Is that OK for you? :-)

HekkiMelody commented:
> > Please don't have README.md and README.rst. It should be README.rst for proper parsing inside Odoo, and you can make use of the fragments inside `readme` folder for letting the pre-commit hook to generate a complete README.rst file.
> 
> Thanks @pedrobaeza ! I have pushed a new `READED.rst` and removed the `.md`. I had issues with pre-commit locally, so I still only have the README.rst file, without the `readme` repo. Is that OK for you? :-)

Even if you have trouble with pre-commit, you can generate a Readme using `oca-gen-addon-readme` which is ultimately what pre-commit does anyway.
https://github.com/OCA/maintainer-tools/?tab=readme-ov-file#readme-generator

stephansainleger commented:
@aleuffre @pedrobaeza I finally manage to propose a README file accepted by pre-commit :sweat_smile: 
All checked being passed, could you please perform final step ?
Thks for your advice! :pray:

pedrobaeza commented:
/ocabot migration partner_contact_in_several_companies

stephansainleger commented:
Hi @pedrobaeza, it seems a review is still expecting from you :-)

pedrobaeza commented:
Hi, I abstain from reviewing, as I don't use this module (nor recommend it). I just did a review on the "form thing", and now I have dismissed it.

njeudy reviewed (approved):
Code review and real test on prod databases

stephansainleger commented:
Hi @pedrobaeza !
It seems all the required checks and reviewed are done. Could you, or another core maintainer, please merge this PR?
Thanks a lot ! :pray:

pedrobaeza commented:
OK, a PSC hasn't reviewed it, but having some old contributors like Nicolas here, let's merge it.

/ocabot merge nobump

pedrobaeza commented:
It seems the fragments syntax is not correct when you combine it in the main README.rst. Check if you are using a title underlining. Only `~~~~~~` is allowed.

stephansainleger commented:
> It seems the fragments syntax is not correct when you combine it in the main README.rst. Check if you are using a title underlining. Only `~~~~~~` is allowed.

Indeed there was wrong underlining. It is fixed! :-)

pedrobaeza commented:
/ocabot merge nobump

stephansainleger commented:
Thanks :pray: :pray:  @pedrobaeza and all others for your kind assistance, this is my first merged PR in OCA, and normally not the last! :muscle: :smile:
